portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -32,22 +40,29 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    
-    ArrayList<String> jsonString = new ArrayList<>();
-    addToList();
-    for(String obj : strings){
-        jsonString.add(convertToJSON(obj));
+
+    Query query = new Query("Comments").addSort("comment", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<String> toSendOver = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+        System.out.println(entity);
+        String title = (String) entity.getProperty("comment");
+        toSendOver.add(title);
     }
 
-    response.setContentType("text/html;");
-    int randomIndex = (int)Math.floor(Math.random() * ((int)jsonString.size()));
-    System.out.print(jsonString.size() + " randome num: " + randomIndex);
-    response.getWriter().println(jsonString.get(randomIndex).toString());
+    Gson gson = new Gson();
+
+    response.setContentType("application/json");
+    response.getWriter().write(gson.toJson(toSendOver));
+    response.sendRedirect("/postRequest.html");
```

Comment:
why redirect, isnt this making it go to another page?

Reply:
This line has been removed